crates/moonbuild/src/entry.rs

```diff
@@ -310,6 +312,75 @@ impl std::fmt::Display for TestResult {
     }
 }
 
+// (pkg's runnable artifacts, <mbt file name, array of (index, test_name)>) for blackbox whitebox internal test
+#[allow(clippy::type_complexity)]
+fn convert_moonc_test_info(
+    test_info_file: &Path,
+    pkg: &Package,
+    output_format: &str,
+    filter_file: Option<&String>,
+) -> anyhow::Result<
+    Vec<(
```

Comment:
define a struct to represent the return type, current is too long and hard to understand